core/src/impl/Kokkos_Default_GraphNodeKernel.hpp

```diff
@@ -95,6 +95,21 @@ struct GraphNodeAggregateKernelDefaultImpl
   void execute_kernel() override final {}
 };
 
+template <typename ExecutionSpace>
+struct GraphNodeThenDefaultImpl
+    : public GraphNodeKernelDefaultImpl<ExecutionSpace> {};
+
+template <typename ExecutionSpace, typename Functor>
+struct GraphNodeThenImpl : public GraphNodeThenDefaultImpl<ExecutionSpace> {
+  Functor m_functor;
+
+  template <typename T>
+  explicit GraphNodeThenImpl(T &&functor)
+      : m_functor(std::forward<T>(functor)) {}
```

Comment:
We discussed about it, and decided it would not be a good idea.
This proposed change was motivated by the default graph implementation. In this implementation, the kernel nodes created with a `Kokkos` parallel construct are executed on the execution space instance given to the policy during definition, such that the user can still "tweak" the default graph implementation to run nodes on different `space`s and effectively get asynchronicity (note that this is because kernel nodes of the default implementation derive from `Kokkos::Impl::Parallel...` structs). But we consider this a hacky hack since it should be the graph itself that decides where its nodes are executed.
So letting the user place a `then` node of the default graph implementation would imply that the following is allowed:
```c++
auto my_node_on_specified_exec = node.then(exec, ...);
```
but it's definitely not a good idea, especially for non-defaulted backends.